internal/command/machine/flaps_api/post.go

```diff
@@ -0,0 +1,189 @@
+		short = "Run a machine from a JSON configuration file"
+		long  = short + "\n"
+
+		usage = "post <json_file> [command]"
+	)
+
+	cmd := command.New(usage, short, long, runPost,
+		command.RequireSession,
+		command.LoadAppNameIfPresent,
+	)
+
+	flag.Add(
+		cmd,
+		flag.Region(),
+		flag.String{
+			Name:        "name",
+			Shorthand:   "n",
+			Description: "Machine name, will be generated if missing",
+		},
+		flag.String{
+			Name:        "org",
+			Description: `The organization that will own the app`,
+		},
+		sharedFlags,
+	)
+
+	cmd.Args = cobra.MinimumNArgs(1)
+
+	return cmd
+}
+
+// Does not take into account any command line flags, such as --app or --name
+func loadMachineJson(ctx context.Context, path string, input *api.LaunchMachineInput) (err error) {
+
+	if !filepath.IsAbs(path) {
+		absPath, err := filepath.Abs(filepath.Join(state.WorkingDirectory(ctx), path))
+		if err != nil {
+			return err
+		}
+		path = absPath
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("failed to open config file: %w", err)
+	}
+
+	defer func() {
+		if e := file.Close(); err == nil {
+			err = e
+		}
+	}()
+
+	if err = json.NewDecoder(file).Decode(input); err != nil {
+
+		return fmt.Errorf("failed parsing machine configuration: %w", err)
+	}
+
+	return nil
+}
+
+func runPost(ctx context.Context) error {
+	var (
+		appName  = app.NameFromContext(ctx)
+		client   = client.FromContext(ctx).API()
+		io       = iostreams.FromContext(ctx)
+		colorize = io.ColorScheme()
+		err      error
+		app      *api.AppCompact
+		input    api.LaunchMachineInput
+	)
+
+	if err = loadMachineJson(ctx, flag.FirstArg(ctx), &input); err != nil {
+		return err
+	}
+
+	if appName == "" {
+		appName = input.AppID
+	}
+
+	if appName == "" {
+		app, err = mach.CreateApp(ctx, "Running a machine without specifying an app will create one for you, is this what you want?", "", client)
+		if err != nil {
+			return err
+		}
+	} else {
+		app, err = client.GetAppCompact(ctx, appName)
+		if err != nil && strings.Contains(err.Error(), "Could not find App") {
+			app, err = mach.CreateApp(ctx, fmt.Sprintf("App '%s' does not exist, would you like to create it?", appName), appName, client)
+			if err != nil {
+				return err
+			}
+			if app == nil {
+				return nil
+			}
+		}
+		if err != nil {
+			return err
+		}
+	}
+	if app == nil {
+		return nil
+	}
+
+	flapsClient, err := flaps.New(ctx, app)
+	if err != nil {
+		return fmt.Errorf("could not make API client: %w", err)
+	}
+	ctx = flaps.NewContext(ctx, flapsClient)
+
+	if app.PlatformVersion == "nomad" {
+		return fmt.Errorf("the app %s uses an earlier version of the platform that does not support machines", app.Name)
+	}
+
+	input.AppID = app.Name
+	// Support a few command-line overrides, but this shouldn't be a clone of `machine run`
+	if name := flag.GetString(ctx, "name"); name != "" {
+		input.Name = name
+	}
+	if region := flag.GetString(ctx, "region"); region != "" {
+		input.Region = region
+	}
```

Comment:
You might be aware of this already and have other reasons for implementing this way, but just in case not: this can be written without the if statement wrapper because the default value for a string in go is `""`:

```suggestion
	input.Region = flag.GetString(ctx, "region")
```

Reply:
I wanted to support region being included in the json, so I only set it when there's an override.